app/actions/websocket/users.ts

```diff
@@ -0,0 +1,26 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {fetchMe} from '@actions/remote/user';
+import DatabaseManager from '@database/manager';
+import {queryCurrentUser} from '@queries/servers/user';
+
+export async function handleUserUpdatedEvent(serverUrl: string, msg: any) {
+    const database = DatabaseManager.serverDatabases[serverUrl];
+    if (!database) {
+        return;
+    }
+    const currentUser = await queryCurrentUser(database.database);
+    const user: UserProfile = msg.data.user;
+
+    if (user.id === currentUser?.id) {
+        if (user.update_at > (currentUser?.updateAt || 0)) {
+            // Need to request me to make sure we don't override with sanitized fields from the
+            // websocket event
+            // TODO Potential improvement https://mattermost.atlassian.net/browse/MM-40582
+            fetchMe(serverUrl, false);
+        }
+    } else {
+        database.operator.handleUsers({users: [user], prepareRecordsOnly: false});
```

Comment:
Ok ok this is great but one thing we need to do here is determine if you have a DM or GM's with this user and ensure the channel display name is updated if needed.

Happy to talk to this over a call to explain a bit more.

Reply:
Done.

I wonder if we should do something similar also if the current user is the one updated, since the locale may have changed (not sure if any other value that goes through this websocket event may trigger any other change here). Or is this somehow handled already by the `fetchMe` function?

Reply:
not sure why the locale is relevant

Reply:
A locale change may change the order of the names in a GM display name (I think, not 100% sure). Not sure if may change also DMs.

Reply:
Sure we can include it just in case and only if the locale has changed